src/lem/circuit.rs

```diff
@@ -514,7 +673,10 @@ impl Func {
                         // a value to the allocated nums to be filled later, we must either
                         // add the results of the call to the witness, or recompute them.
                         let output_vals = if let Some(true) = not_dummy.get_value() {
-                            g.call_outputs.pop_front().unwrap()
+                            g.call_outputs.pop_front().unwrap_or_else(|| {
+                                let dummy = Ptr::Leaf(Tag::Expr(Nil), F::ZERO);
```

Comment:
Under what conditions, in this `not_dummy == true` branch, can `dummy` values be required?

Reply:
This is a very ugly hack which I want to address later. It's just that `call_outputs` should only be populated when we're synthesizing with proper frames, not with blank ones (for shapes). But `not_dummy` always start with `Boolean::Constant(true)`, even in the blank case. A better solution would be starting `not_dummy` as None in the blank case